sdk/eventgrid/eventgrid/src/cloudEventAdapter.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { SendCloudEventInput } from "./models";
+
+interface MessageWithMetadata {
+  /**
+   * The message's binary data
+   */
+  body: Uint8Array;
+  /**
+   * The message's content type
+   */
+  contentType: string;
+}
+
+interface MessageAdapter<MessageT> {
+  /**
+   * defines how to create a message from a payload and a content type
+   */
+  produceMessage: (messageWithMetadata: MessageWithMetadata) => MessageT;
```

Comment:
I was wondering whether there's case where either `produceMessage` or `consumeMessage` could be optional, i.e., no adapting needed.

Reply:
I believe the idea is for these interfaces to be implemented by our packages and not our customers and we will define these two methods for every type we support AFAICT to support both serialization and de-serialization.

Also, if it turns out there is a customer ask for such case, perhaps we can do the optional change easily since going from required to optional should not be a breaking change. What do you think?

Reply:
Makes sense!